src/sentry/api/endpoints/organization_release_details.py

```diff
@@ -186,6 +189,14 @@ def put(self, request, organization, version):
                         datetime=release.date_released,
                     )
 
+            analytics.record(
+                "release.set_commits",
+                user_id=request.user.id if request.user and request.user.id else None,
```

Comment:
@NisanthanNanthakumar I don't think it's possible for `request.user` to be missing. Every authenticated request should be tied to a user.

Reply:
@scefali in redash, if you run 
```
select * from analytics.events where type = "release.created" and JSON_EXTRACT_SCALAR(data, '$.user_id') is null limit 100
```
there are many entries. I don't know if the CLI's auth_token maps to a user

Reply:
@NisanthanNanthakumar But wouldn't the code have errored out earlier if there was no `request.user`? And CLI auth tokens do have an associated user (the proxy user).

Reply:
@scefali at the point where we want to collect analytics, we don't check if `request.user.is_authenticated()`. Also now that this has been refactored into a function in a mixin that can be used anywhere, I think its good to have sane defaults when accessing nested objects incase of undefined.